chore: migration guide for v10 (#1419)
AmadeusITGroup/otter: migration-guides/10.0.md

# Major 10.0 Migration guide
> [!IMPORTANT]
> Please make sure to first follow the ["Upgrade new version guide"](https://github.com/AmadeusITGroup/otter/blob/release/main/docs/core/UPGRADE_NEW_VERSION.md) before going through these steps.

## Personalization

> From version 10 of @o3r, the components' types must be set with the ```@O3rComponent``` decorator instead of the following

Reviewer: ```suggestion
Starting from version 10 of @o3r, the types of the components types must be set with the ```@O3rComponent``` decorator instead of the following
```

> interfaces and decorator (already deprecated) that have been fully removed.

Reviewer: ```suggestion
previously deprecated interfaces that have been fully removed.
```

If you have not yet done so, the migration to ``@ama-sdk/schematics`` is automated since v9:

```shell
# In an angular context
ng add @ama-sdk/schematics

# yarn usage
yarn add -D @angular-devkit/schematics-cli @ama-sdk/schematics
yarn schematics @ama-sdk/schematics:ng-add

# npm usage
npm install --save-dev @ama-sdk/schematics @angular-devkit/schematics-cli
npx -p @angular-devkit/schematics-cli schematics @ama-sdk/schematics:ng-add
```

## Rules engine split
The previous ``RulesEngineModule`` from ``@o3r/rules-engine`` has been split into a runner (``RulesEngineRunnerModule``
available in ``@o3r/rules-engine``) and dedicated modules to handle the actions triggered by the engine:
* ``AssetRulesEngineActionModule`` available in ``@o3r/dynamic-content``
* ``ConfigurationRulesEngineActionModule`` available in ``@o3r/configuration``
* ``LocalizationRulesEngineActionModule`` available in ``@o3r/localization``
* ``PlaceholderRulesEngineActionModule`` available in ``@o3r/component``

This change allows a better tuning of the stores and packages loaded with your application. This means that from version

> 10, actions need to be explicitly registered in your application. 

Reviewer: ```suggestion
10 onwards, actions need to be explicitly registered in your application. 
```
